### PR TITLE
Add embed type and autoplay settings for Soundcloud

### LINKED ISSUE #96: Add shortcode for Soundcloud
It should also reverse this:

```
<iframe width="100%" height="450" scrolling="no" frameborder="no" src="https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/208170049&amp;auto_play=false&amp;hide_related=false&amp;show_comments=true&amp;show_user=true&amp;show_reposts=false&amp;visual=true"></iframe>
```

### PR DESCRIPTION
**Pull request by danielbachhuber**
Fixes #96

### CODE REVIEW

**Comment by danielbachhuber** [comment]
@goldenapples #reviewmerge

File: tests/test-soundcloud-shortcode.php [hunk 1]
```diff
@@ -14,6 +14,12 @@ public function test_post_display_api_url() {
 		$this->assertContains( '<iframe width="100%" height="166" scrolling="no" frameborder="no" src="https://w.soundcloud.com/player/?url=https%3A%2F%2Fapi.soundcloud.com%2Ftracks%2F219074591"></iframe>', apply_filters( 'the_content', $post->post_content ) );
 	}
 
+	public function test_post_display_autoplay_visual() {
+		$post_id = $this->factory->post->create( array( 'post_content' => '[soundcloud url="https://soundcloud.com/wondalandarts/hell-you-talmbout" type="visual" autoplay="1"]' ) );
+		$post = get_post( $post_id );
+		$this->assertContains( '<iframe width="100%" height="350" scrolling="no" frameborder="no" src="https://w.soundcloud.com/player/?url=https%3A%2F%2Fsoundcloud.com%2Fwondalandarts%2Fhell-you-talmbout&#038;visual=true&#038;auto_play=true"></iframe>', apply_filters( 'the_content', $post->post_content ) );
```
Comment (goldenapples) [inline tests/test-soundcloud-shortcode.php]: Why is this height 350, when you set it to 450 in the callback?

**Comment by goldenapples** [comment]
@danielbachhuber Can you fix the tests? Looks good other than that height number.

File: tests/test-soundcloud-shortcode.php (hunk 1 above)
Comment (danielbachhuber) [inline tests/test-soundcloud-shortcode.php]: Herp a derp. Because I bash at my keyboard at times.

**Pull request merged**